packages/proverif/proverif.1.98/opam

```diff
@@ -0,0 +1,32 @@
+opam-version: "1.2"
+maintainer:
+  "Bruno Blanchet <bruno.blanchet@inria.fr>, Marc Sylvestre <marc.sylvestre@inria.fr>"
+authors:
+  "Bruno Blanchet <bruno.blanchet@inria.fr>, Vincent Cheval <vincent.cheval@icloud.com>, Ben Smyth <research@bensmyth.com>, Marc Sylvestre <marc.sylvestre@inria.fr>"
+homepage: "http://proverif.inria.fr/"
+bug-reports: "bruno.blanchet@inria.fr"
+license: "GNU General Public License"
+build: [
+       [ "./build" "ocb.byte" ] { !ocaml-native }
+       [ "./build" "ocb.native" ] { ocaml-native }
+]
+install: [ "./build" "install" "%{prefix}%" ]
+
+remove: [
+  [ "rm" "-rf"
+      "%{prefix}%/doc/proverif"
+      "%{prefix}%/bin/proverif"
+      "%{prefix}%/bin/proveriftotex"
+      "%{prefix}%/bin/proverif_interact"
+      "%{prefix}%/bin/proverif.exe"
+      "%{prefix}%/bin/proveriftotex.exe"
+      "%{prefix}%/bin/proverif_interact.exe"
+  ]
+]
+depends: ["ocamlfind"
+          "ocamlbuild"
+	  "lablgtk"]
+depexts: [
+  [[] ["graphviz"] ]
```

Comment:
This should be forbidden. This field requires you to at least put the name of the packages for each distributions.
Anyway, there is a ```conf-graphviz``` package already so this should be removed and replaced by a dependency over this packages.

Reply:
Thanks! I have made the change you suggested.

Reply:
Now the Alpine 3.6 build fails because conf-graphviz fails on this distribution...
Would adding a line [["alpine"] ["graphviz"]] in the depexts field of the opam file of the conf-graphviz package fix the problem?
By the way, the CentOS build also fails; I do not understand why. It may not be related to the ProVerif package. If you understand why, I would be happy to know. Thanks for your help.

Reply:
The CentOS image is broken. This is not caused by this PR, it's totally ""normal"".
Yes it would. Could you add it to the ```conf-graphviz``` package ?

Reply:
On a second thought, I am going to remove the dependency on conf-graphviz entirely, for the following reasons:
- The dependency is optional: proverif works without graphviz. So it could go into depopts.
- But there is no need to build conf-graphviz before proverif, or to recompile proverif if conf-graphviz changes, so it is useless to put it into depopts.
The best behavior in my opinion would be to try to install graphviz, but still continue in case of failure. I do not know how to do that with opam.

I could fix conf-graphviz, but wouldn't it be better if the maintainer of that package fixes it?
I plan to email him unless you tell me otherwise.

Reply:
> But there is no need to build conf-graphviz before proverif, or to recompile proverif if conf-graphviz changes, so it is useless to put it into depopts.
The best behavior in my opinion would be to try to install graphviz, but still continue in case of failure. I do not know how to do that with opam.

Using the ```build``` attribute over ```conf-graphviz``` and putting it into depopts should do the trick I suppose.

Reply:
Really? As far as I understand, opam does not install what is in depopts, so that would not try to install graphviz. And in fact, graphviz is not needed for building proverif, it is needed only to run it (for some features only), so using the "build" attribute seems strange. I do not see what that would improve with respect to current package (with no dependency on conf-graphviz).
Could you please accept the package as it is? It works fine. Thanks a lot.

Reply:
Ok, merging. The ```build``` attribute just tells opam to not rebuild the package if the ```conf-graphviz``` is being upgraded.